pkg/workloadmeta/collectors/internal/containerd/container_builder.go

```diff
@@ -100,3 +111,40 @@ func extractStatus(status containerd.ProcessStatus) workloadmeta.ContainerStatus
 
 	return workloadmeta.ContainerStatusUnknown
 }
+
+// extractIP gets the IP of a container.
+//
+// The containerd client does not expose the IPs, that's why we use the helpers
+// that we have in the "system" package to extract that information from
+// "/proc".
+//
+// A current limitation is that if a container exposes multiple IPs, this
+// function just returns one of them. That means that if a container is attached
+// to multiple networks this might not work as expected.
+func extractIP(container containerd.Container, containerdClient cutil.ContainerdItf) (string, error) {
+	taskPids, err := containerdClient.TaskPids(container)
```

Comment:
IIRC, this call is costly and is also used in metrics provider, maybe it should be checked again and see if we need to build a cache (there's already one in metrics provider I believe)